### ISSUE TITLE
4 bpp PNGs not supported: Unexpected colourType: 3

### ISSUE THREAD

**Issue by tgrajewski** [image-js/image-js#402]
Would be great if this package supported more color types, including 4 bits per pixel in PNGs.

Currently I get this error when loading such images in Node:
```
> (node:11512) UnhandledPromiseRejectionWarning: Error: Unexpected colourType: 3
    at loadPNG (image-js/lib/image/core/load.js:123:13)
    at loadBinary (image-js/lib/image/core/load.js:66:16)
    at binaryDataP.then.binaryData (/image-js/lib/image/core/load.js:95:12)
```
Example 4 bpp image:
![circle](https://user-images.githubusercontent.com/4324821/45222741-99905d80-b2b5-11e8-87b0-b5f6da409470.png)

**Comment by targos** [comment]
If I'm not mistaken, colourType: 3 means that the colours are based on a palette. I'm definitely open to supporting such images but the question is what to do with them? Convert to regular 32bpp?
Adding direct support for colour palettes to the library is an option but that would take ages.

**Comment by tgrajewski** [comment]
Yes, upgrading to 32 bpp should be fine.

**Comment by targos** [comment]
This should be fixed in version 0.21.1 ([commit](https://github.com/image-js/image-js/commit/f6042f18df81f55889a4ce875eb31f5c35e727b2).
I added a test with your image and another one with a palette saved from Photoshop.

**Comment by targos** [comment]
@tgrajewski Can we close this issue?
